crates/primitives/src/signature/sig.rs

```diff
@@ -13,7 +13,7 @@ const SECP256K1N_ORDER: U256 =
     uint!(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141_U256);
 
 /// An Ethereum ECDSA signature.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, Hash, Default, PartialEq, Eq)]
 pub struct Signature {
```

Comment:
I don't think we should add this. IMO it's not great in general to have `Default` impl producing invalid type. we already have `Signature::test_signature` which can be used in tests

Reply:
We'll need to remove `Default` on a few type encapsulating `Signature` on reth: 

* BlobTransaction
* TransactionSigned
* TransactionSignedNoHash
* TransactionSignedEcRecovered

I was expecting more than that, seems doable.

Reply:
feel free to keep those as manual impls, I think we only use them for testing, just don't want to get this into core